src/enemies/test_zombie/TestZombie.gd

```diff
@@ -92,10 +92,22 @@ func take_damage(data: DamageData) -> void:
 		return
 	
 	if data.does_source_aggro():
-		target = data.source
+		aggro(data.source)
 		if alert_level < 0.5:
 			alert_level = 0.5
 
+## Function that handles aggroing and de-aggroing. Adds itself to the aggro_array if its targeting a Player.
+## Removes itself from the aggro_array otherwise.
+func aggro(tar: CharacterBody2D) -> void:
+	if tar != target:
+		target = tar
+		if target is Player:
+			EnemyUtil.aggro_array.append(self)
+		elif EnemyUtil.aggro_array.has(self):
+			EnemyUtil.aggro_array.erase(self)
```

Comment:
i don't like this direct access to the aggro array. what if we *did* switch to a dictionary? we'd need to change all these accesses to that.